src/core/server/http/http_server.ts

```diff
@@ -323,17 +331,32 @@ export class HttpServer {
     this.server.events.on('response', this.handleServerResponseEvent);
   }
 
-  private setupRequestStateAssignment(config: HttpConfig) {
+  private setupRequestStateAssignment(
+    config: HttpConfig,
+    executionContext?: InternalExecutionContextSetup
+  ) {
     this.server!.ext('onRequest', (request, responseToolkit) => {
+      const requestId = getRequestId(request, config.requestId);
+
+      const parentContext = executionContext?.getParentContextFrom(request.headers);
```

Comment:
It can be extracted to a single method, but it's okay as long as we have this logic isolated in the Core.